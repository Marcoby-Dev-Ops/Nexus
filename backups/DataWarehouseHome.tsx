import React from 'react';
import { Database, Search, Download, Upload, BarChart2, Users, FileText, Filter } from 'lucide-react';
import { PageTemplates } from '@/components/patterns/PageTemplates';
import { ContentCard } from '@/components/patterns/ContentCard';
import { KpiCard } from '@/components/dashboard/KpiCard';
import { SimpleBarChart } from '@/components/dashboard/SimpleBarChart';

/**
 * @name DataWarehouseHome
 * @description Data warehouse dashboard for accessing and analyzing business data.
 * @returns {JSX.Element} The rendered DataWarehouseHome component.
 */

// Sample data for the data warehouse dashboard
const dataKpiData = [
  { title: 'Total Records', value: '2.4M', delta: '+15.2%', trend: 'up' },
  { title: 'Data Sources', value: 12, delta: '+2', trend: 'up' },
  { title: 'Storage Used', value: '1.2TB', delta: '+8.5%', trend: 'up' },
  { title: 'Query Performance', value: '98.5%', delta: '+1.2%', trend: 'up' },
];

const usageData = [
  { name: 'Sales Data', value: 850 },
  { name: 'Customer Data', value: 650 },
  { name: 'Financial Data', value: 420 },
  { name: 'Operational Data', value: 380 },
  { name: 'Marketing Data', value: 320 },
];

const queryData = [
  { name: 'Mon', value: 45 },
  { name: 'Tue', value: 52 },
  { name: 'Wed', value: 38 },
  { name: 'Thu', value: 67 },
  { name: 'Fri', value: 85 },
  { name: 'Sat', value: 23 },
  { name: 'Sun', value: 18 },
];

const dataSources = [
  { name: 'Sales CRM', type: 'Database', status: 'Connected', lastSync: '2 minutes ago', records: '450K' },
  { name: 'E-commerce Platform', type: 'API', status: 'Connected', lastSync: '5 minutes ago', records: '1.2M' },
  { name: 'Financial System', type: 'CSV', status: 'Syncing', lastSync: '1 hour ago', records: '180K' },
  { name: 'Marketing Tools', type: 'API', status: 'Connected', lastSync: '10 minutes ago', records: '95K' },
];

const recentQueries = [
  { query: 'Monthly Sales Report', user: 'John Doe', time: '5 minutes ago', duration: '2.3s' },
  { query: 'Customer Segmentation Analysis', user: 'Jane Smith', time: '15 minutes ago', duration: '4.1s' },
  { query: 'Revenue Trend Analysis', user: 'Mike Johnson', time: '30 minutes ago', duration: '1.8s' },
  { query: 'Product Performance Metrics', user: 'Sarah Wilson', time: '45 minutes ago', duration: '3.2s' },
];

const DataWarehouseHome: React.FC = () => {
  return (
    <PageTemplates.Department
      title="Data Warehouse"
      subtitle="Access and analyze your business data with powerful querying and visualization tools"
    >
      {/* Quick Actions */}
      <ContentCard title="Quick Actions" variant="elevated" className="mb-8">
        <div className="grid grid-cols-2 md:grid-cols-4 gap-4">
          <button className="flex flex-col items-center p-6 rounded-lg bg-background border border-border hover:border-primary hover:bg-primary/5 hover:shadow-md transition-all duration-200 group">
            <div className="p-3 rounded-lg bg-primary/10 text-primary group-hover:bg-primary/20 transition-all duration-200 mb-3">
              <Search className="w-5 h-5" />
            </div>
            <span className="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200">
              Query Data
            </span>
          </button>
          
          <button className="flex flex-col items-center p-6 rounded-lg bg-background border border-border hover:border-primary hover:bg-primary/5 hover:shadow-md transition-all duration-200 group">
            <div className="p-3 rounded-lg bg-primary/10 text-primary group-hover:bg-primary/20 transition-all duration-200 mb-3">
              <Upload className="w-5 h-5" />
            </div>
            <span className="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200">
              Import Data
            </span>
          </button>
          
          <button className="flex flex-col items-center p-6 rounded-lg bg-background border border-border hover:border-primary hover:bg-primary/5 hover:shadow-md transition-all duration-200 group">
            <div className="p-3 rounded-lg bg-primary/10 text-primary group-hover:bg-primary/20 transition-all duration-200 mb-3">
              <Download className="w-5 h-5" />
            </div>
            <span className="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200">
              Export Report
            </span>
          </button>
          
          <button className="flex flex-col items-center p-6 rounded-lg bg-background border border-border hover:border-primary hover:bg-primary/5 hover:shadow-md transition-all duration-200 group">
            <div className="p-3 rounded-lg bg-primary/10 text-primary group-hover:bg-primary/20 transition-all duration-200 mb-3">
              <BarChart2 className="w-5 h-5" />
            </div>
            <span className="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200">
              Create Dashboard
            </span>
          </button>
        </div>
      </ContentCard>

      {/* KPI Cards */}
      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6 mb-8">
        {dataKpiData.map((kpi, index) => (
          <KpiCard key={index} {...kpi} />
        ))}
      </div>

      {/* Charts */}
      <div className="grid grid-cols-1 lg:grid-cols-2 gap-8 mb-8">
        <ContentCard 
          title="Data Usage by Category" 
          variant="elevated"
          className="hover:shadow-xl transition-all duration-300"
        >
          <div className="mb-6">
            <p className="text-sm text-muted-foreground">Storage usage across data categories</p>
          </div>
          <SimpleBarChart data={usageData} />
        </ContentCard>

        <ContentCard 
          title="Query Activity" 
          variant="elevated"
          className="hover:shadow-xl transition-all duration-300"
        >
          <div className="mb-6">
            <p className="text-sm text-muted-foreground">Daily query volume (last 7 days)</p>
          </div>
          <SimpleBarChart data={queryData} />
        </ContentCard>
      </div>

      {/* Data Sources and Recent Queries */}
      <div className="grid grid-cols-1 lg:grid-cols-2 gap-8">
        <ContentCard 
          title="Data Sources" 
          variant="elevated"
          action={
            <button className="px-4 py-4 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 text-sm font-medium border border-border">
              Manage Sources
            </button>
          }
        >
          <div className="space-y-4">
            {dataSources.map((source, index) => (
              <div key={index} className="flex items-center justify-between p-4 rounded-lg hover:bg-muted/50 transition-colors duration-200">
                <div className="flex items-center space-x-4">
                  <div className="p-4 rounded-lg bg-primary/10">
                    <Database className="w-5 h-5 text-primary" />
                  </div>
                  <div>
                    <h4 className="font-medium text-foreground">{source.name}</h4>
                    <p className="text-sm text-muted-foreground">{source.type} • {source.records} records</p>
                  </div>
                </div>
                <div className="text-right">
                  <div className={`inline-flex px-4 py-4 rounded-full text-xs font-medium ${
                    source.status === 'Connected' ? 'bg-success/10 text-success' :
                                                    source.status === 'Syncing' ? 'bg-warning/10 text-warning-foreground' :
                    'bg-destructive/10 text-destructive'
                  }`}>
                    {source.status}
                  </div>
                  <p className="text-xs text-muted-foreground mt-1">{source.lastSync}</p>
                </div>
              </div>
            ))}
          </div>
        </ContentCard>

        <ContentCard 
          title="Recent Queries" 
          variant="elevated"
          action={
            <button className="px-4 py-4 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 text-sm font-medium border border-border">
              Query History
            </button>
          }
        >
          <div className="space-y-4">
            {recentQueries.map((query, index) => (
              <div key={index} className="flex items-center justify-between p-4 rounded-lg hover:bg-muted/50 transition-colors duration-200">
                <div className="flex items-center space-x-4">
                  <div className="p-4 rounded-lg bg-primary/10">
                    <Search className="w-5 h-5 text-primary" />
                  </div>
                  <div>
                    <h4 className="font-medium text-foreground">{query.query}</h4>
                    <p className="text-sm text-muted-foreground">by {query.user} • {query.time}</p>
                  </div>
                </div>
                <div className="text-right">
                  <div className="text-sm font-medium text-foreground">{query.duration}</div>
                </div>
              </div>
            ))}
          </div>
        </ContentCard>
      </div>
    </PageTemplates.Department>
  );
};

export default DataWarehouseHome; 