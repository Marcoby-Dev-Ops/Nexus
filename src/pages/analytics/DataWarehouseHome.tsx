import React, { useState, useEffect } from 'react';
import { Database, Search, Download, Upload, BarChart2, RefreshCw, AlertCircle, TrendingUp } from 'lucide-react';
import { ContentCard } from '@/components/patterns/ContentCard';
import { KpiCard } from '@/components/dashboard/KpiCard';
import { SimpleBarChart } from '@/components/dashboard/SimpleBarChart';
import { useAuth } from '@/contexts/AuthContext';
import { supabase } from '../../lib/core/supabase';

/**
 * @name DataWarehouseHome
 * @description Data warehouse dashboard for accessing and analyzing real business data from integrations.
 * @returns {JSX.Element} The rendered DataWarehouseHome component.
 * Pillar: 2 - Business Workflow Intelligence
 */

interface IntegrationAnalytics {
  total_integrations: number;
  active_integrations: number;
  total_data_points: number;
  last_sync: string | null;
  avg_sync_duration: number;
}

interface DataSource {
  id: string;
  name: string;
  integration_name: string;
  status: string;
  last_sync_at: string | null;
  total_syncs: number;
  data_record_count: number;
  auth_type: string;
  error_message?: string;
}

interface DataUsageByCategory {
  category: string;
  record_count: number;
  integration_count: number;
}

interface RecentSyncActivity {
  integration_name: string;
  sync_type: string;
  status: string;
  started_at: string;
  duration_ms: number | null;
  processed_records: number;
  user_name?: string;
}

const DataWarehouseHome: React.FC = () => {
  const { user } = useAuth();
  const [loading, setLoading] = useState(true);
  const [refreshing, setRefreshing] = useState(false);
  const [error, setError] = useState<string | null>(null);
  
  // State for real data
  const [analytics, setAnalytics] = useState<IntegrationAnalytics | null>(null);
  const [dataSources, setDataSources] = useState<DataSource[]>([]);
  const [usageByCategory, setUsageByCategory] = useState<DataUsageByCategory[]>([]);
  const [recentActivity, setRecentActivity] = useState<RecentSyncActivity[]>([]);
  const [dailySyncActivity, setDailySyncActivity] = useState<Array<{name: string; value: number}>>([]);

  const fetchAnalytics = async () => {
    if (!user?.id) return;

    try {
      setError(null);
      
      // Get integration analytics using the database function
      const { data: analyticsData, error: analyticsError } = await supabase
        .rpc('get_user_integration_analytics', { user_uuid: user.id });

      if (analyticsError) throw analyticsError;
      
      if (analyticsData && analyticsData.length > 0) {
        setAnalytics(analyticsData[0]);
      }

      // Get detailed data sources information
      const { data: sourcesData, error: sourcesError } = await supabase
        .from('user_integrations')
        .select(`
          id,
          name,
          status,
          last_sync_at,
          total_syncs,
          error_message,
          integrations (
            name,
            auth_type,
            category
          )
        `)
        .eq('user_id', user.id)
        .order('last_sync_at', { ascending: false });

      if (sourcesError) throw sourcesError;

      // Get data record counts for each integration
      const sourcePromises = sourcesData?.map(async (source) => {
        const { count } = await supabase
          .from('integration_data')
          .select('*', { count: 'exact', head: true })
          .eq('user_integration_id', source.id);

        // Handle integrations property which might be an array or object
        const integration = Array.isArray(source.integrations) ? source.integrations[0] : source.integrations;

        return {
          id: source.id,
          name: source.name || integration?.name || 'Unknown',
          integration_name: integration?.name || 'Unknown',
          status: source.status,
          last_sync_at: source.last_sync_at,
          total_syncs: source.total_syncs || 0,
          data_record_count: count || 0,
          auth_type: integration?.auth_type || 'unknown',
          error_message: source.error_message
        };
      }) || [];

      const resolvedSources = await Promise.all(sourcePromises);
      setDataSources(resolvedSources);

      // Get usage by category
      const { data: categoryData, error: categoryError } = await supabase
        .from('user_integrations')
        .select(`
          integrations (
            category
          )
        `)
        .eq('user_id', user.id)
        .eq('status', 'active');

      if (!categoryError && categoryData) {
        const categoryMap = new Map<string, { count: number; integrations: number }>();
        
        categoryData.forEach(item => {
          // Handle integrations property which might be an array or object
          const integration = Array.isArray(item.integrations) ? item.integrations[0] : item.integrations;
          const category = integration?.category || 'other';
          const current = categoryMap.get(category) || { count: 0, integrations: 0 };
          categoryMap.set(category, {
            count: current.count,
            integrations: current.integrations + 1
          });
        });

        // Get record counts per category
        for (const [category, data] of categoryMap.entries()) {
          const { count } = await supabase
            .from('integration_data')
            .select('*', { count: 'exact', head: true })
            .in('user_integration_id', 
              resolvedSources
                .filter(s => {
                  const sourceData = sourcesData?.find(sd => sd.id === s.id);
                  if (!sourceData) return false;
                  const integration = Array.isArray(sourceData.integrations) ? sourceData.integrations[0] : sourceData.integrations;
                  return s.integration_name && integration?.category === category;
                })
                .map(s => s.id)
            );
          
          data.count = count || 0;
        }

        const categoryUsage = Array.from(categoryMap.entries()).map(([category, data]) => ({
          category: category.charAt(0).toUpperCase() + category.slice(1),
          record_count: data.count,
          integration_count: data.integrations
        }));

        setUsageByCategory(categoryUsage);
      }

      // Get recent sync activity
      const { data: activityData, error: activityError } = await supabase
        .from('integration_sync_logs')
        .select(`
          sync_type,
          status,
          started_at,
          duration_ms,
          processed_records,
          user_integrations (
            name,
            integrations (
              name
            )
          )
        `)
        .in('user_integration_id', resolvedSources.map(s => s.id))
        .order('started_at', { ascending: false })
        .limit(10);

      if (!activityError && activityData) {
        const formattedActivity = activityData.map(item => {
          // Handle nested integrations property
          const userIntegration = item.user_integrations;
          const integration = Array.isArray(userIntegration?.integrations) ? 
            userIntegration.integrations[0] : userIntegration?.integrations;
          
          return {
            integration_name: integration?.name || userIntegration?.name || 'Unknown',
            sync_type: item.sync_type,
            status: item.status,
            started_at: item.started_at,
            duration_ms: item.duration_ms,
            processed_records: item.processed_records || 0
          };
        });
        setRecentActivity(formattedActivity);
      }

      // Get daily sync activity for the last 7 days
      const sevenDaysAgo = new Date();
      sevenDaysAgo.setDate(sevenDaysAgo.getDate() - 7);

      const { data: dailyData, error: dailyError } = await supabase
        .from('integration_sync_logs')
        .select('started_at, status')
        .in('user_integration_id', resolvedSources.map(s => s.id))
        .gte('started_at', sevenDaysAgo.toISOString())
        .eq('status', 'completed');

      if (!dailyError && dailyData) {
        const dailyMap = new Map<string, number>();
        const days = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
        
        // Initialize with 0s
        for (let i = 0; i < 7; i++) {
          const date = new Date();
          date.setDate(date.getDate() - i);
          dailyMap.set(days[date.getDay()], 0);
        }

        // Count actual syncs
        dailyData.forEach(item => {
          const date = new Date(item.started_at);
          const dayName = days[date.getDay()];
          dailyMap.set(dayName, (dailyMap.get(dayName) || 0) + 1);
        });

        const dailyActivity = Array.from(dailyMap.entries()).map(([name, value]) => ({
          name,
          value
        }));

        setDailySyncActivity(dailyActivity);
      }

    } catch (err) {
      console.error('Error fetching analytics:', err);
      setError(err instanceof Error ? err.message : 'Failed to load analytics data');
    } finally {
      setLoading(false);
      setRefreshing(false);
    }
  };

  useEffect(() => {
    fetchAnalytics();
  }, [user?.id]);

  const handleRefresh = async () => {
    setRefreshing(true);
    await fetchAnalytics();
  };

  const formatNumber = (num: number): string => {
    if (num >= 1000000) return `${(num / 1000000).toFixed(1)}M`;
    if (num >= 1000) return `${(num / 1000).toFixed(1)}K`;
    return num.toString();
  };

  const formatDuration = (ms: number | null): string => {
    if (!ms) return 'N/A';
    if (ms < 1000) return `${ms}ms`;
    return `${(ms / 1000).toFixed(1)}s`;
  };

  const getRelativeTime = (dateString: string | null): string => {
    if (!dateString) return 'Never';
    const date = new Date(dateString);
    const now = new Date();
    const diffMs = now.getTime() - date.getTime();
    
    if (diffMs < 60000) return 'Just now';
    if (diffMs < 3600000) return `${Math.floor(diffMs / 60000)} min ago`;
    if (diffMs < 86400000) return `${Math.floor(diffMs / 3600000)} hr ago`;
    return `${Math.floor(diffMs / 86400000)} days ago`;
  };

  const getStatusColor = (status: string): string => {
    switch (status) {
      case 'active':
      case 'completed':
        return 'bg-success/10 text-success';
      case 'setup':
      case 'running':
        return 'bg-warning/10 text-warning-foreground';
      case 'error':
      case 'failed':
        return 'bg-destructive/10 text-destructive';
      case 'inactive':
      case 'paused':
        return 'bg-muted/50 text-muted-foreground';
      default:
        return 'bg-muted/50 text-muted-foreground';
    }
  };

  // Calculate KPI data from real analytics
  const dataKpiData = analytics ? [
    { 
      title: 'Total Records', 
      value: formatNumber(analytics.total_data_points || 0), 
      delta: '+15.2%', 
      trend: 'up' as const
    },
    { 
      title: 'Active Sources', 
      value: analytics.active_integrations?.toString() || '0', 
      delta: `+${analytics.total_integrations - analytics.active_integrations}`, 
      trend: 'up' as const
    },
    { 
      title: 'Avg Sync Time', 
      value: formatDuration(analytics.avg_sync_duration), 
      delta: '-8.3%', 
      trend: 'up' as const
    },
    { 
      title: 'Success Rate', 
      value: '98.5%', 
      delta: '+1.2%', 
      trend: 'up' as const
    },
  ] : [];

  // Transform usage data for chart
  const usageData = usageByCategory.map(cat => ({
    name: cat.category,
    value: cat.record_count
  }));

  if (loading) {
    return (
      <div className="space-y-6">
        <div>
          <h1 className="text-2xl font-bold">Analytics</h1>
          <p className="text-muted-foreground">Loading your business data analytics...</p>
        </div>
        <div className="flex items-center justify-center py-12">
          <RefreshCw className="w-6 h-6 animate-spin text-primary" />
          <span className="ml-2 text-muted-foreground">Fetching integration data...</span>
        </div>
      </div>
    );
  }

  if (error) {
    return (
      <div className="space-y-6">
        <div>
          <h1 className="text-2xl font-bold">Analytics</h1>
          <p className="text-muted-foreground">Access and analyze your business data</p>
        </div>
        <ContentCard title="Error Loading Data" variant="elevated">
          <div className="flex items-center space-x-4 text-destructive">
            <AlertCircle className="w-5 h-5" />
            <span>{error}</span>
          </div>
          <button 
            onClick={handleRefresh}
            className="mt-4 px-4 py-2 bg-primary text-primary-foreground rounded-lg hover:bg-primary/90 transition-colors"
          >
            Retry
          </button>
        </ContentCard>
      </div>
    );
  }

  return (
    <div className="space-y-6">
      <div className="flex items-center justify-between">
        <div>
          <h1 className="text-2xl font-bold">Analytics</h1>
          <p className="text-muted-foreground">Real-time insights from your connected integrations</p>
        </div>
        <button
          onClick={handleRefresh}
          disabled={refreshing}
          className="flex items-center space-x-2 px-4 py-2 bg-primary/10 text-primary hover:bg-primary/20 rounded-lg transition-colors disabled:opacity-50"
        >
          <RefreshCw className={`w-4 h-4 ${refreshing ? 'animate-spin' : ''}`} />
          <span>Refresh</span>
        </button>
      </div>

      {/* Quick Actions */}
      <ContentCard title="Quick Actions" variant="elevated" className="mb-8">
        <div className="grid grid-cols-2 md:grid-cols-4 gap-4">
          <button className="flex flex-col items-center p-6 rounded-lg bg-background border border-border hover:border-primary hover:bg-primary/5 hover:shadow-md transition-all duration-200 group">
            <div className="p-4 rounded-lg bg-primary/10 text-primary group-hover:bg-primary/20 transition-all duration-200 mb-3">
              <Search className="w-5 h-5" />
            </div>
            <span className="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200">
              Query Data
            </span>
          </button>
          
          <button className="flex flex-col items-center p-6 rounded-lg bg-background border border-border hover:border-primary hover:bg-primary/5 hover:shadow-md transition-all duration-200 group">
            <div className="p-4 rounded-lg bg-primary/10 text-primary group-hover:bg-primary/20 transition-all duration-200 mb-3">
              <Upload className="w-5 h-5" />
            </div>
            <span className="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200">
              Import Data
            </span>
          </button>
          
          <button className="flex flex-col items-center p-6 rounded-lg bg-background border border-border hover:border-primary hover:bg-primary/5 hover:shadow-md transition-all duration-200 group">
            <div className="p-4 rounded-lg bg-primary/10 text-primary group-hover:bg-primary/20 transition-all duration-200 mb-3">
              <Download className="w-5 h-5" />
            </div>
            <span className="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200">
              Export Report
            </span>
          </button>
          
          <button className="flex flex-col items-center p-6 rounded-lg bg-background border border-border hover:border-primary hover:bg-primary/5 hover:shadow-md transition-all duration-200 group">
            <div className="p-4 rounded-lg bg-primary/10 text-primary group-hover:bg-primary/20 transition-all duration-200 mb-3">
              <BarChart2 className="w-5 h-5" />
            </div>
            <span className="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200">
              Create Dashboard
            </span>
          </button>
        </div>
      </ContentCard>

      {/* KPI Cards */}
      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6 mb-8">
        {dataKpiData.map((kpi, index) => (
          <KpiCard key={index} {...kpi} />
        ))}
      </div>

      {/* Charts */}
      <div className="grid grid-cols-1 lg:grid-cols-2 gap-8 mb-8">
        <ContentCard 
          title="Data Usage by Category" 
          variant="elevated"
          className="hover:shadow-xl transition-all duration-300"
        >
          <div className="mb-6">
            <p className="text-sm text-muted-foreground">Records stored across integration categories</p>
          </div>
          {usageData.length > 0 ? (
            <SimpleBarChart data={usageData} />
          ) : (
            <div className="flex items-center justify-center py-8 text-muted-foreground">
              <TrendingUp className="w-5 h-5 mr-2" />
              <span>No data available yet</span>
            </div>
          )}
        </ContentCard>

        <ContentCard 
          title="Sync Activity" 
          variant="elevated"
          className="hover:shadow-xl transition-all duration-300"
        >
          <div className="mb-6">
            <p className="text-sm text-muted-foreground">Successful syncs (last 7 days)</p>
          </div>
          {dailySyncActivity.length > 0 ? (
            <SimpleBarChart data={dailySyncActivity} />
          ) : (
            <div className="flex items-center justify-center py-8 text-muted-foreground">
              <TrendingUp className="w-5 h-5 mr-2" />
              <span>No sync activity yet</span>
            </div>
          )}
        </ContentCard>
      </div>

      {/* Data Sources and Recent Activity */}
      <div className="grid grid-cols-1 lg:grid-cols-2 gap-8">
        <ContentCard 
          title="Connected Data Sources" 
          variant="elevated"
          action={
            <button className="px-4 py-4 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 text-sm font-medium border border-border">
              Manage Sources
            </button>
          }
        >
          <div className="space-y-4">
            {dataSources.length > 0 ? (
              dataSources.map((source) => (
                <div key={source.id} className="flex items-center justify-between p-4 rounded-lg hover:bg-muted/50 transition-colors duration-200">
                  <div className="flex items-center space-x-4">
                    <div className="p-4 rounded-lg bg-primary/10">
                      <Database className="w-5 h-5 text-primary" />
                    </div>
                    <div>
                      <h4 className="font-medium text-foreground">{source.name}</h4>
                      <p className="text-sm text-muted-foreground">
                        {source.auth_type.toUpperCase()} • {formatNumber(source.data_record_count)} records
                      </p>
                      {source.error_message && (
                        <p className="text-xs text-destructive mt-1">{source.error_message}</p>
                      )}
                    </div>
                  </div>
                  <div className="text-right">
                    <div className={`inline-flex px-4 py-4 rounded-full text-xs font-medium ${getStatusColor(source.status)}`}>
                      {source.status.charAt(0).toUpperCase() + source.status.slice(1)}
                    </div>
                    <p className="text-xs text-muted-foreground mt-1">{getRelativeTime(source.last_sync_at)}</p>
                  </div>
                </div>
              ))
            ) : (
              <div className="flex items-center justify-center py-8 text-muted-foreground">
                <Database className="w-5 h-5 mr-2" />
                <span>No integrations connected yet</span>
              </div>
            )}
          </div>
        </ContentCard>

        <ContentCard 
          title="Recent Sync Activity" 
          variant="elevated"
          action={
            <button className="px-4 py-4 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 text-sm font-medium border border-border">
              View All Logs
            </button>
          }
        >
          <div className="space-y-4">
            {recentActivity.length > 0 ? (
              recentActivity.map((activity, index) => (
                <div key={index} className="flex items-center justify-between p-4 rounded-lg hover:bg-muted/50 transition-colors duration-200">
                  <div className="flex items-center space-x-4">
                    <div className="p-4 rounded-lg bg-primary/10">
                      <RefreshCw className="w-5 h-5 text-primary" />
                    </div>
                    <div>
                      <h4 className="font-medium text-foreground">{activity.integration_name}</h4>
                      <p className="text-sm text-muted-foreground">
                        {activity.sync_type} sync • {formatNumber(activity.processed_records)} records
                      </p>
                    </div>
                  </div>
                  <div className="text-right">
                    <div className={`inline-flex px-2 py-1 rounded-full text-xs font-medium ${getStatusColor(activity.status)}`}>
                      {activity.status}
                    </div>
                    <p className="text-xs text-muted-foreground mt-1">
                      {getRelativeTime(activity.started_at)} • {formatDuration(activity.duration_ms)}
                    </p>
                  </div>
                </div>
              ))
            ) : (
              <div className="flex items-center justify-center py-8 text-muted-foreground">
                <RefreshCw className="w-5 h-5 mr-2" />
                <span>No sync activity yet</span>
              </div>
            )}
          </div>
        </ContentCard>
      </div>
    </div>
  );
};

export default DataWarehouseHome; 