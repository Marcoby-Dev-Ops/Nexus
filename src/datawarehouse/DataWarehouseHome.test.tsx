/**
 * @file DataWarehouseHome.test.tsx
 * @description Unit and snapshot tests for the DataWarehouseHome component.
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import DataWarehouseHome from './DataWarehouseHome';

describe('DataWarehouseHome', () => {
  it('renders data warehouse title', () => {
    render(<DataWarehouseHome />);
    expect(screen.getByText(/data warehouse/i)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<DataWarehouseHome />);
    expect(asFragment()).toMatchSnapshot();
  });
}); 