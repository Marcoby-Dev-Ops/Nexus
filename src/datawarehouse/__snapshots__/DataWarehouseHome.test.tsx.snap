// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataWarehouseHome matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-8 space-y-8"
  >
    <div
      class="grid grid-cols-1 md:grid-cols-4 gap-4"
    >
      <div
        class="rounded-xl border p-4 shadow-sm flex flex-col gap-1"
      >
        <p
          class="text-muted-foreground mb-1 text-xs uppercase"
        >
          Total Records
        </p>
        <div
          class="flex items-end justify-between gap-2"
        >
          <p
            class="text-3xl font-semibold"
          >
            12450
          </p>
        </div>
      </div>
      <div
        class="rounded-xl border p-4 shadow-sm flex flex-col gap-1"
      >
        <p
          class="text-muted-foreground mb-1 text-xs uppercase"
        >
          Data Sources
        </p>
        <div
          class="flex items-end justify-between gap-2"
        >
          <p
            class="text-3xl font-semibold"
          >
            5
          </p>
        </div>
      </div>
      <div
        class="rounded-xl border p-4 shadow-sm flex flex-col gap-1"
      >
        <p
          class="text-muted-foreground mb-1 text-xs uppercase"
        >
          Last Sync
        </p>
        <div
          class="flex items-end justify-between gap-2"
        >
          <p
            class="text-3xl font-semibold"
          >
            2024-06-10
          </p>
        </div>
      </div>
      <div
        class="rounded-xl border p-4 shadow-sm flex flex-col gap-1"
      >
        <p
          class="text-muted-foreground mb-1 text-xs uppercase"
        >
          Active Reports
        </p>
        <div
          class="flex items-end justify-between gap-2"
        >
          <p
            class="text-3xl font-semibold"
          >
            8
          </p>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl border p-4"
    >
      <h3
        class="font-semibold mb-2"
      >
        Data Explorer
      </h3>
      <table
        class="min-w-full text-sm"
      >
        <thead>
          <tr>
            <th
              class="text-left p-2"
            >
              Type
            </th>
            <th
              class="text-left p-2"
            >
              Name
            </th>
            <th
              class="text-left p-2"
            >
              Amount
            </th>
            <th
              class="text-left p-2"
            >
              Date
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="border-t"
          >
            <td
              class="p-2"
            >
              Invoice
            </td>
            <td
              class="p-2"
            >
              INV-001
            </td>
            <td
              class="p-2"
            >
              $2,500
            </td>
            <td
              class="p-2"
            >
              2024-06-10
            </td>
          </tr>
          <tr
            class="border-t"
          >
            <td
              class="p-2"
            >
              Deal
            </td>
            <td
              class="p-2"
            >
              Acme Corp
            </td>
            <td
              class="p-2"
            >
              $5,000
            </td>
            <td
              class="p-2"
            >
              2024-06-09
            </td>
          </tr>
          <tr
            class="border-t"
          >
            <td
              class="p-2"
            >
              Expense
            </td>
            <td
              class="p-2"
            >
              AWS
            </td>
            <td
              class="p-2"
            >
              $200
            </td>
            <td
              class="p-2"
            >
              2024-06-08
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 gap-4"
    >
      <div
        class="rounded-xl border p-4"
      >
        <h3
          class="font-semibold mb-2"
        >
          Revenue Trend
        </h3>
        <div
          class="recharts-responsive-container"
          style="width: 100%; height: 200px; min-width: 0;"
        />
      </div>
      <div
        class="rounded-xl border p-4"
      >
        <h3
          class="font-semibold mb-2"
        >
          Records by Month
        </h3>
        <div
          class="recharts-responsive-container"
          style="width: 100%; height: 200px; min-width: 0;"
        />
      </div>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 gap-4"
    >
      <div
        class="rounded-xl border p-4"
      >
        <h3
          class="font-semibold mb-2"
        >
          AI Insights
        </h3>
        <ul
          class="list-disc pl-5"
        >
          <li>
            No anomalies detected in the last sync.
          </li>
          <li>
            Revenue trend is up 8% over last quarter.
          </li>
        </ul>
      </div>
      <div
        class="rounded-xl border p-4 mb-8"
      >
        <h3
          class="font-semibold mb-2"
        >
          Quick Actions
        </h3>
        <div
          class="flex gap-2 flex-wrap"
        >
          <button
            class="bg-primary text-white rounded px-4 py-2"
          >
            Export Data
          </button>
          <button
            class="bg-secondary text-black rounded px-4 py-2"
          >
            Connect Source
          </button>
          <button
            class="bg-muted text-black rounded px-4 py-2"
          >
            Run Report
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
